### PR TITLE
Hack 20: Create sub bullet lists inside a Rich Text element

### PR DESCRIPTION
**Pull request by brianndiiri**

### CODE REVIEW

**Review by alexiglesias93 — CHANGES_REQUESTED** [review]
This solution is CSS based, but it's actually not valid HTML syntax.
Check this:
https://developer.mozilla.org/en-US/docs/Learn/HTML/Introduction_to_HTML/HTML_text_fundamentals#nesting_lists

You should nest the sub-bullets by creating an additional list element.

File: webflow-hacks/misc-hacks/20/20.ts [hunk 1]
```diff
@@ -0,0 +1,39 @@
+document.addEventListener('DOMContentLoaded', function () {
+  const UL_SELECTOR = '[fs-hacks-element="unordered-list"]';
+  const originalList = document.querySelector(UL_SELECTOR);
+  const unorderedList = document.createElement('ul');
+  if (!originalList) return;
+
+  const listItems = originalList.querySelectorAll('li');
+
+  for (const item of listItems) {
+    const tildeCount = (item.innerText.match(/~/g) || []).length;
```
Comment (alexiglesias93) [inline webflow-hacks/misc-hacks/20/20.ts]: I think we should only match `~` at the beginning of the text. Currently, you're checking for `~` in the entire strings, which means that this string:
> ~ Joe ate ~ 4 apples approximately.
Would count as 2 tildes, when in reality it should only be one.

Same appleis to the `cleanListItem` function, it should only clean the `~` at the beginning of the strings.

File: webflow-hacks/misc-hacks/20/20.ts [hunk 2]
```diff
@@ -0,0 +1,39 @@
+document.addEventListener('DOMContentLoaded', function () {
+  const UL_SELECTOR = '[fs-hacks-element="unordered-list"]';
+  const originalList = document.querySelector(UL_SELECTOR);
+  const unorderedList = document.createElement('ul');
```
Comment (alexiglesias93) [inline webflow-hacks/misc-hacks/20/20.ts]: What about ordered lists? No support for them?

**Review by alexiglesias93 — CHANGES_REQUESTED** [review]

File: webflow-hacks/misc-hacks/20/20.ts (hunk 2 above)
Comment (brianndiiri) [reply webflow-hacks/misc-hacks/20/20.ts]: Perhaps a new hack for this?

File: webflow-hacks/misc-hacks/20/20.ts (hunk 1 above)
Comment (brianndiiri) [reply webflow-hacks/misc-hacks/20/20.ts]: In a list within a list, tildes can appear on the same line. So there are many cases where matching on the beginning of the line would render an invalid list but few cases where a semantic tilde would bring list formatting issue. Please check https://finsweetwebflowhacks-rewrite.webflow.io/20

File: webflow-hacks/misc-hacks/20/20.ts (hunk 2 above)
Comment (alexiglesias93) [reply webflow-hacks/misc-hacks/20/20.ts]: Hm... feels like we would be repeating too much code.
What would be the implementation of nested ordered lists?
How would the nested bullets be represented? Numbers? Letters? We need more insights.

Comment (brianndiiri) [reply webflow-hacks/misc-hacks/20/20.ts]: How about 1,2,3, for top level list item, then 3.1, 3.2 for sub bullets, then 3.1.1 for sub-sub bullet

Comment (alexiglesias93) [reply webflow-hacks/misc-hacks/20/20.ts]: Works for me!

**Review by alexiglesias93 — APPROVED** [review]

**Pull request merged**